impl/src/main/java/org/ehcache/internal/store/offheap/AbstractOffHeapStore.java

```diff
@@ -832,44 +833,67 @@ public boolean test(Map.Entry<K, OffHeapValueHolder<V>> argument) {
 
   class OffHeapStoreIterator implements Iterator<Cache.Entry<K, ValueHolder<V>>> {
     private final java.util.Iterator<Map.Entry<K, OffHeapValueHolder<V>>> mapIterator;
+    private Map.Entry<K, OffHeapValueHolder<V>> next = null;
 
     OffHeapStoreIterator() {
       mapIterator = backingMap().entrySet().iterator();
+      advance();
+    }
+
+    private void advance() {
+      next = null;
+      while (next == null && mapIterator.hasNext()) {
+        Map.Entry<K, OffHeapValueHolder<V>> entry = mapIterator.next();
+        final long now = timeSource.getTimeMillis();
+        if (entry.getValue().isExpired(now, TimeUnit.MILLISECONDS)) {
+          continue;
```

Comment:
Exactly. Removal of those entries will happen in subsequent get/compute calls. That should be fine, right?